src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/filtering/Mutect2FilteringEngine.java

```diff
@@ -189,9 +194,40 @@ public VariantContext applyFiltersAndAccumulateOutputStats(final VariantContext
             }
         }
 
+        List<String> siteFilters = new ArrayList<>();
+        List<Map<Allele, String>> newFilters =
+                errorProbabilities.getProbabilitiesByFilterAndAllele().entrySet().stream().map(
+                entry -> addFilterStrings(entry.getValue(), siteFilters, entry.getKey().filterName())).collect(Collectors.toList());
+
+        siteFilters.forEach(vcb::filter);
+        List<String> orderedASFilterStrings = vc.getAlleles().stream().map(allele -> getMergedFilterString(allele, newFilters)).collect(Collectors.toList());
+        String finalAttrString = AnnotationUtils.encodeAnyASList(orderedASFilterStrings);
+
+        vcb.putAttributes(Collections.singletonMap(GATKVCFConstants.AS_FILTER_STATUS_KEY, finalAttrString));
         return vcb.make();
     }
 
+    private String getMergedFilterString(Allele allele, List<Map<Allele, String>> newFilters) {
```

Comment:
Some Javadoc here would be nice, for posterity, but it's up to you since these are `private`.